### ISSUE TITLE
Semantics is misaligned with actual component on web

### ISSUE THREAD

**Issue by Harrisonnguyen97** [flutter/flutter#171264]
### Steps to reproduce

1. Enable semantics in flutter web 
2. Add any long data table with many items to enable scrolling 
3. Start to scroll fast

### Expected results

Semantics nodes are correctly aligned with the items

### Actual results

- First row of the datatable is always misaligned whether scrolling or not 
=> Remove dataRowMinHeight will fix this problem, but I want to be able to set this value
- When scrolling fast, semantics nodes are misaligned with the items

### Code sample
```
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: DataTableScreen(),
    );
  }
}

class DataTableScreen extends StatelessWidget {
  final List<Map<String, dynamic>> users = [
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
    {'name': 'Alice', 'age': 30, 'email': 'alice@example.com'},
    {'name': 'Bob', 'age': 25, 'email': 'bob@example.com'},
    {'name': 'Charlie', 'age': 35, 'email': 'charlie@example.com'},
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('User Data Table')),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(16.0),
        scrollDirection: Axis.vertical,
        child: DataTable(
          dataRowMaxHeight: double.infinity,
          dataRowMinHeight: 70,
          showCheckboxColumn: false,
          columns: const [
            DataColumn(label: Text('Name')),
            DataColumn(label: Text('Age')),
            DataColumn(label: Text('Email')),
          ],
          rows: users
              .map(
                (user) => DataRow(
                  cells: [
                    DataCell(
                      onTap: () => print(user['name']),
                      Text(user['name']),
                    ),
                    DataCell(Text(user['age'].toString())),
                    DataCell(Text(user['email'])),
                  ],
                ),
              )
              .toList(),
        ),
      ),
    );
  }
}
```

### Screenshots or Video

https://github.com/user-attachments/assets/c067a2cc-3b98-4dc2-bcf0-366217884eff

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor 
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.32.4, on macOS 14.6.1 23G93 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.101.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate the reported behavior with latest sdk versions.

```
stable : 3.32.4
master : 3.33.0-1.0.pre.748
```

**Comment by mdebbar** [comment]
The cells of the first row have a `transform` with a `y` component that's non zero for some reason. Since the cell has the same vertical position as the row it belongs to, the `y` component of the cell should be zero.

Cells of other rows are fine; their transforms have a zero `y` component.

These transforms are sent from the framework during semantics updates, so this is most likely a framework issue.
